### PR TITLE
refactor(provider/amazon): convert createApplicationLoadBalancer to TS

### PR DESCRIPTION
**Pull request by jrsquared**
In an effort to keep this small, just converting createApplicationLoadBalancer first and making minimal changes needed to createClassicLoadBalancer to support the new shape expected by the templates.

Next will be to convert createClassicLoadBalancer and extract out the common bits.

Also snuck in typing of all the `$uibModalInstance`s

### CODE REVIEW

File: app/scripts/modules/amazon/src/loadBalancer/configure/application/createApplicationLoadBalancer.controller.ts [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+  TASK_MONITOR_BUILDER,
+  TaskMonitorBuilder,
+  TaskMonitor,
+  V2_MODAL_WIZARD_SERVICE
+} from '@spinnaker/core';
+
+import { AWSProviderSettings } from 'amazon/aws.settings';
+import { IALBListener,
+  IAmazonApplicationLoadBalancer,
+  IAmazonApplicationLoadBalancerUpsertCommand
```
Comment (anotherchrisberry) [inline app/scripts/modules/amazon/src/loadBalancer/configure/application/createApplicationLoadBalancer.controller.ts]: (not suggesting a change in this PR, just something to discuss...)

Should we introduce a "command" or "ops" package to separate domain data and operations?

**Comment by anotherchrisberry** [comment]
This is a very large PR so LGTM

File: app/scripts/modules/amazon/src/loadBalancer/configure/application/createApplicationLoadBalancer.controller.ts (hunk 1 above)
Comment (jrsquared) [inline app/scripts/modules/amazon/src/loadBalancer/configure/application/createApplicationLoadBalancer.controller.ts]: Great idea.

**Comment by jrsquared** [comment]
wcgw

**Pull request merged**